core/services/pipeline/task.length.go

```diff
@@ -0,0 +1,43 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// Return types:
+//
+//	*decimal.Decimal
+type LengthTask struct {
+	BaseTask `mapstructure:",squash"`
+	Input    string `json:"input"`
+}
+
+var _ Task = (*LengthTask)(nil)
+
+func (t *LengthTask) Type() TaskType {
+	return TaskTypeLength
+}
+
+func (t *LengthTask) Run(_ context.Context, _ logger.Logger, vars Vars, inputs []Result) (result Result, runInfo RunInfo) {
+	_, err := CheckInputs(inputs, 0, 1, 0)
+	if err != nil {
+		return Result{Error: errors.Wrap(err, "task inputs")}, runInfo
+	}
+
+	var input BytesParam
+
+	err = multierr.Combine(
+		errors.Wrap(ResolveParam(&input, From(VarExpr(t.Input, vars), NonemptyString(t.Input), Input(inputs, 0))), "input"),
+	)
+	if err != nil {
+		return Result{Error: err}, runInfo
+	}
+
+	return Result{Value: decimal.NewFromInt(int64(len(input)))}, runInfo
```

Comment:
What if input is a json array string? e.g. [1, 2, 3], the user would expect to get 3. Did you consider this case?

Reply:
not yet, I think this can be generalised for multiple things in iterations. For now I made the behaviour explicit in docs.